### DISCUSSION TITLE
Community Review: BEP022 - MR Spectroscopy

### DISCUSSION THREAD

**Discussion by effigies** [bids-standard/bids-specification#1846] [BEP Review]
We are pleased to announce the community review period for BIDS Extension Proposal (BEP) 022!

BEP022 extends the BIDS standard to include Magnetic Resonance Spectroscopy (MRS). We have developed the extension to align with the current community consensus on MRS data and metadata terminology and organization while also matching as closely as possible and appropriately with the BIDS’s guidelines for MRI techniques.

The draft specification may be found at: https://bids-specification.readthedocs.io/en/bep022/. This is kept up-to-date with the [latest](https://bidsschematools.readthedocs.io/en/latest/) branch of the specification with frequent merges.

The proposed changes may be found at https://github.com/bids-standard/bids-specification/pull/1377. To view the source difference, click the [Files changed](https://github.com/bids-standard/bids-specification/pull/1377/files) tab.

Example datasets may be found at:

* [mrs_2dmrsi](https://github.com/markmikkelsen/bids-examples/tree/bep022/mrs_2dmrsi)
* [mrs_biggaba](https://github.com/markmikkelsen/bids-examples/tree/bep022/mrs_biggaba)
* [mrs_fmrs](https://github.com/markmikkelsen/bids-examples/tree/bep022/mrs_fmrs)

# Procedure

## Review period

Community review is open from June 3 - June 14, 2024.

## Voting

This discussion is a poll, which you can vote in once and cannot change your response. Therefore, please vote once you are satisfied that your answer will not change.

If you vote "No", you may create a thread to explain your reasoning or send a private email to bids.maintenance [at] gmail.com so that your position can be fully understood.

## How to review

Use threads in this discussion to request clarification or changes to the BEP. Each independent topic or proposed change should have its own thread. Please review existing threads and consider adding comments to those, if your topics are already being discussed. If your concerns are adequately represented by an existing comment or comments, consider using the 👍🏻 reaction button to indicate agreement in place of extending the discussion.

Where possible, propose the specific text you would like to see added or removed.

For minor issues, such as typos, feel free to make code suggestions only, using the [suggest feature](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#adding-comments-to-a-pull-request) on the [BEP022 pull request](https://github.com/bids-standard/bids-specification/pull/1377/files). These suggestions may be accepted or rejected without discussion, so *do not* make substantial proposals in this way.

## Finalization

Once the review period is finished, the BEP leads and maintainers will consider the entire discussion, identify outstanding points to be resolved, and work toward finalizing the BEP. Community members may be asked to participate in this process to provide the necessary expertise.

Before being merged, the BEP will be integrated into the BIDS validator, and the [examples](https://github.com/bids-standard/bids-examples/pull/425) must pass the validator.

Once merged, the specification will be frozen for a final review period of 1 week before the release of BIDS 1.10.0.

If, for any reason, a critical issue is found after the BEP is merged, the BEP will be reverted while the issues are addressed. At the discretion of the BIDS steering group and maintainers group, depending on the scope of changes, a new community review period may be opened.

---

## Note

The use of GitHub discussions, and especially polling, is new to this BEP. The goal is to make the process more legible than past reviews, where pull requests became a mix of typo suggestions, comment threads on lines of code, and unthreaded discussions on the main page. We may choose to do something else in the future, or even scrap this and try again with this review. We appreciate your patience.

**Comment by effigies** [comment]
# Announcements and amendments

Because updating the main post resets votes, any updates that do not justify resetting votes will be posted to this thread.

**Comment by ctr** [comment]
Hi @wtclarke and others. I hope this is the right place to comment.

Would you consider adding a "methodDoi" header to link to the DOI for the main paper describing the acquisition approach used? This may be a helpful historical record to bake into MRS data as there are so many one-off special methods. It could then aid an understanding of how the other parameters are (ab)used for new sequences beyond what is anticipated today.

Can the JSON note whether the MRS spatial coordinates are defined with or without distortion correction? I presume normally it's without distortion correction but not sure that is always the case. This will otherwise be a source of mis-alignment.

Does the specification offer guidance as to how to complete the Vendor, etc labels in cases of simulated data?

Does it provide a space to specify samples that should be dropped before analysis, e.g. initial points before a spin echo or initial points that are known to be corrupted during acquisition?

Thanks

Chris Rodgers.

**Reply by wtclarke** [reply]
@markmikkelsen should definitely get most of the credit, I'm definitely in the `other` pile here.

Good point on the  first three issues, but I'll leave those with greater knowledge of how BIDs works to answer. On the final point, this is something I've been meaning to (but forgetting to) add to NIfTI-MRS. I've made an issue on the NIfTI-MRS repo to discuss what implementation to use, see https://github.com/wtclarke/mrs_nifti_standard/issues/12.

**Reply by effigies** [reply]
> consider adding a "methodDoi" header to link to the DOI

> Does it provide a space to specify samples that should be dropped before analysis, e.g. initial points before a spin echo or initial points that are known to be corrupted during acquisition?

These questions seem to be targeted at NIfTI-MRS, not the BIDS-MRS specification, but there is overlap here, as the BIDS sidecar could in principle contain anything that is in the NIfTI-MRS header extension (both being JSON). Perhaps it's worth taking this opportunity to think about how to evolve the specs in parallel.

Both specifications permit unknown metadata fields, to allow users to include whatever they think is important. So I could imagine a process like:

1. Researchers define their own metadata as needed
2. Researchers propose well-defined fields for inclusion in the NIfTI-MRS specification
3. NIfTI-MRS / BIDS contributors propose an equivalent field for BIDS sidecar, with an explicit link to the MRS name
4. NIfTI-MRS and/or BIDS maintain a machine-readable table of NIfTI-MRS fields and their BIDS equivalents, so a tool could straightforwardly load the NIfTI-MRS header and the BIDS sidecar and sync up the metadata.

Note that it is difficult to add new required metadata to BIDS too often, as that breaks backwards compatibility. But if NIfTI-MRS headers are versioned, that can evolve separately, and BIDS can maintain "MUST NOT conflict" rules rather than "REQUIRED" rules until a major version bump allows backwards compatibility to be dropped.

----

> Can the JSON note whether the MRS spatial coordinates are defined with or without distortion correction? I presume normally it's without distortion correction but not sure that is always the case. This will otherwise be a source of mis-alignment.

Is this susceptibility distortion or gradient nonlinearity that you're considering? [NonlinearGradientCorrection](https://bids-specification.readthedocs.io/en/stable/glossary.html#nonlineargradientcorrection-metadata) exists and could be added to the MRS metadata. It is currently "RECOMMENDED, but REQUIRED if PET data are present" for MRI, but MRS could have a different requirement level. We do not have a way of indicating susceptibility distortion correction, but I see no reason not to add that if it's useful.

----

> Does the specification offer guidance as to how to complete the Vendor, etc labels in cases of simulated data?

No, and I would personally prefer that we not get into that business in the BIDS spec itself, which is already quite large. There is plenty of room for recommending conventions for particular use cases, and I believe many of those have gone into the [Starter Kit](https://bids-standard.github.io/bids-starter-kit/) in the past. @Remi-Gau would probably remember better.

**Reply by wtclarke** [reply]
> consider adding a "methodDoi" header to link to the DOI
>>These questions seem to be targeted at NIfTI-MRS, not the BIDS-MRS specification

`methodDoi`, or an equivalent link to some primary source, would definitely have much wider application than just MRS (though MRS would be a big beneficiary). Many of the more complex/newer/exploratory contrasts (e.g. ASL, CEST, parameter mapping etc) are littered with special or evolving methods. Is this something that could be suggested as a generic key in the BIDs standard?

**Reply by effigies** [reply]
> Is this something that could be suggested as a generic key in the BIDs standard?

I see no reason not to propose it. At present, you can always include sources at the dataset level in `ReferencesAndLinks`.

I would suggest proposing it after this BEP is released, however, as you never know when a broad proposal will go through quickly or result in a lot of arguing.

**Comment by ywpark84** [comment]
Quick question. Would the use of the dielectric padding be considered a "B1ShimmingTechnique"?

**Reply by markmikkelsen** [reply]
I would say yes. `B1ShimmingTechnique` takes strings, so you can be as descriptive as you want in describing the technique used.

**Comment by effigies** [comment]
The review period is over. There are no objections, and so we will prepare BEP022 for release in BIDS 1.10.0.

We anticipate BIDS 1.10.0 to be released by the end of August, 2024, contingent upon completion of validation and examples.

**Reply by markmikkelsen** [reply]
Thanks to everyone who took the time to review! Much appreciated!
